src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Triggers/EventHubAsyncCollector.cs

```diff
@@ -45,25 +45,25 @@ public EventHubAsyncCollector(EventHubClient client)
         /// <param name="item">The event to add</param>
         /// <param name="cancellationToken">a cancellation token. </param>
         /// <returns></returns>
-        public async Task AddAsync(EventData item, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task AddAsync(PartitionedValue item, CancellationToken cancellationToken = default(CancellationToken))
         {
             if (item == null)
             {
                 throw new ArgumentNullException("item");
             }
 
-            string key = item.SystemProperties?.PartitionKey ?? string.Empty;
+            string key = item.PartitionKey ?? string.Empty;
 
             PartitionCollector partition;
             lock (_partitions)
             {
                 if (!_partitions.TryGetValue(key, out partition))
                 {
-                    partition = new PartitionCollector(this);
+                    partition = new PartitionCollector(this, key);
                     _partitions[key] = partition;
                 }
             }
-            await partition.AddAsync(item, cancellationToken);
+            await partition.AddAsync(item.EventData, cancellationToken);
         }
 
         /// <summary>
```

Comment:
Would it make sense to have one more overload for AddAsync, where you can pass an EventData and a partition key? It saves the user the step of having to create the PartitionedValue if they are binding to EventHubAsyncCollector. That way most C# users won't need to learn about the PartitionedValue type that we've introduced.

Reply:
In fact if we do this, it might also make sense to make IAsyncCollector<EventData>.AddAsync an explicit interface implementation, so it only appears in intellisense if you're binding to IAsyncCollector (and not EventHubAsyncCollector).

Reply:
Looking through the host , here is the code that binds to AsyncCollector for primitive types: 
https://github.com/Azure/azure-functions-host/blob/e709f1d32f2941d06ae4aa9eed195789969a028b/src/WebJobs.Script/Binding/ExtensionBinding.cs#L49-L60

Note: this takes only data for the binding via [context.value](https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Binding/FunctionBinding.cs#L147)

This would require introducing a new type that encapsulates data and partition key. 
 
I think it would be more useful to add AddAsync that takes a dictionary of properties
```
Task AddAsync(T item, Dictionary<string, object> OutpuDataProperties, CancellationToken cancellationToken = default(CancellationToken));
```
This would allow setting any other properties that SDKs allow on the ouput bindings

Reply:
Adding @mathewc in case he has better ideas of handling data and properties on the output binding.

Reply:
If we are using the same hashed paritionkey, is this guaranteed to  be sent to the same eventub partition?  I do not know if we are using the same EventHub client/connection or not. Plus if this will effect the partitionkey mapping.

Reply:
One more question if  the user created more than 32 keys. are we going to hash two values to the same partition?

Reply:
@Pragna I think Pragna is pointing out that these changes only address this issue for .NET, not out of proc languages. OOP languages will be binding to string value only and have no way to specify these additional properties.

This is issue has been around from the beginning for OOP languages, and we have various tracking items, e.g. https://github.com/Azure/azure-functions-host/issues/364 for the case of blobs/Content-Type.

Our position on this has always been that if you need to set any advanced properties on the output value, just use the native SDK directly. If we decide that isn't sufficient and we want to attempt to enable OOP lanugages to specify additional properties that map to the .NET native type, we have some feature work to do in the binding pipeline, as I suggested in https://github.com/Azure/azure-functions-host/issues/364 (e.g. support for dataType: object binding, which we'd then map to the native type like EventData, etc.)

Reply:
@alrod Is there away for mapping the additional data and pass it along with the data (byte[]) as a JObject?